src/marshmallow_annotations/builder.py

```diff
@@ -0,0 +1,236 @@
+
+        kwargs.setdefault('allow_none', allow_none)
+        kwargs.setdefault('required', required)
+
+        field_constructor = self.registry.from_type(typehint)
+        return field_constructor(self, subtypes, kwargs)
+
+    def _exists_on_scheme(self, name):
+        if self.scheme is None:
+            return False
+
+    def _get_meta_options(self, name):
+        if self.scheme is None:
+            return {}
+
+        parent_meta_options = self._visit_parent_metas(
+            lambda meta: getattr(meta, name, None)
+        )
+
+        return parent_meta_options or {}
+
+    def _visit_parent_metas(self, f):
+        for parent in self.scheme.__mro__:
+            meta = getattr(parent, 'Meta', None)
+            if meta is not None:
+                result = f(meta)
+                if result:
+                    return result
+
+    def __call__(self, typehint, **k):
+        return self._field_from_typehint(typehint, k)
+
+    def _get_type_hints(self, item):
+        hints = {}
+
+        for parent in item.__mro__[::-1]:
+            hints.update(get_type_hints(parent))
+
+        return hints
+
+    def convert(self, name, typehint, **kwargs):
+        opts = {**self._get_meta_options(name), **kwargs}
+        return self._field_from_typehint(typehint, opts)
+
+    def convert_all(self, type, ignore=frozenset()):
+        return {
+            k: self.convert(k, v)
+            for k, v in self._get_type_hints(type).items() if k not in ignore
+        }
+
+
+class AnnotationSchemaMeta(SchemaMeta):
+
+    @staticmethod
+    def __new__(mcls, name, bases, attrs, **k):
+        cls = super().__new__(mcls, name, bases, attrs)
+        meta = getattr(cls, 'Meta', None)
+        cls._register_as_scheme_for_target(meta)
+        return cls
+
+    @classmethod
+    def get_declared_fields(
+            mcls, klass, cls_fields, inherited_fields, dict_cls
+    ):
+        fields = super().get_declared_fields(
+            klass, cls_fields, inherited_fields, dict_cls
+        )
+
+        meta = getattr(klass, 'Meta', None)
+
+        if not meta:
+            return fields
+
+        target = getattr(meta, 'target', None)
+
+        if target is None:
+            return fields
+
+        converter_factory = getattr(meta, 'converter', BaseConverter)
+
+        # weeeee circular references!
+        meta.converter = converter = converter_factory(klass)
+
+        # ignore anything explicitly declared on this scheme
+        # or any parent scheme
+        ignore = fields.keys()
+        fields.update(converter.convert_all(target, ignore))
+
+        return fields
+
+    @classmethod
+    def _register_as_scheme_for_target(cls, meta):
+        if meta is None:
+            return
+
+        should_register = getattr(meta, 'register_as_scheme', False)
+        target = getattr(meta, 'target', None)
+
+        if should_register and target:
+            TypeRegistry.register_scheme_constructor(target, cls)
```

Comment:
Converter accepts a type registry argument -- should this put it there or on the global one?